xbmc/dbwrappers/mysqldataset.cpp

```diff
@@ -435,7 +435,33 @@ int MysqlDatabase::drop_analytics(void) {
       if ( (ret=query_with_reconnect(sql)) != MYSQL_OK )
       {
         mysql_free_result(res);
-        throw DbErrors("Can't create trigger '%s'\nError: %d", row[0], ret);
+        throw DbErrors("Can't drop trigger '%s'\nError: %d", row[0], ret);
+      }
+    }
+    mysql_free_result(res);
+  }
+
+  // Native functions
+  sprintf(sql,
+          "SELECT routine_name "
+          "FROM information_schema.routines "
+          "WHERE routine_type = 'FUNCTION' and routine_schema = '%s'",
+          db.c_str());
```

Comment:
Is it safer to use `snprintf` instead of `sprintf`?

Reply:
Use of sprintf is consistent with the rest of the routine